Sources/protoc-gen-grpc-swift/Generator-Client+AsyncAwait.swift

```diff
@@ -130,6 +130,115 @@ extension Generator {
   }
 }
 
+// MARK: - Client protocol extension: Safe call wrappers.
+
+extension Generator {
+  internal func printAsyncClientProtocolSafeWrappersExtension() {
+    self.printAvailabilityForAsyncAwait()
+    self.withIndentation("extension \(self.asyncClientProtocolName)", braces: .curly) {
+      // 'Safe' calls.
+      for (i, method) in self.service.methods.enumerated() {
+        if i > 0 {
+          self.println()
+        }
+        self.method = method
+
+        let rpcType = streamingType(self.method)
+        let callTypeWithoutPrefix = Types.call(for: rpcType, withGRPCPrefix: false)
+        let responseStreamType = "GRPCAsyncResponseStream"
+
+        switch rpcType {
+        case .unary:
+          self.printFunction(
+            name: self.methodFunctionName,
+            arguments: [
+              "_ request: \(self.methodInputName)",
+              "callOptions: \(Types.clientCallOptions)? = nil",
+            ],
+            returnType: self.methodOutputName,
+            access: self.access,
+            async: true,
+            throws: true
+          ) {
+            self.withIndentation(
+              "return try await self.perform\(callTypeWithoutPrefix)",
+              braces: .round
+            ) {
+              self.println("path: \(self.methodPath),")
+              self.println("request: request,")
+              self.println("callOptions: callOptions ?? self.defaultCallOptions")
+            }
+          }
+
+        case .serverStreaming:
```

Comment:
This is the same as `.unary`, I think?

Reply:
You are partially right! This wasn't the case before we factored the code into the `GRPCClient` extension which was probably why I missed it, but there's more similarity now. However there are still some notable differences:

* whether the input is generic
* input name param: request vs requests
* return type: response vs. response stream
* effects: `async throws` vs none
* return statement requiring `try await` vs not

What I have done is created commit a62c2e96 which refactors this. Do you prefer it or does it add too much indirection to read the codegen clearly?

Reply:
Ah, I missed that -- I'm happy with how it is now.